typescript/sdk/src/governance/app.ts

```diff
@@ -1,136 +1,50 @@
-import { ethers } from 'ethers';
-
+import { GovernanceContracts } from '.';
 import { AbacusApp } from '../app';
-import { domains } from '../domains';
-import { ChainName } from '../types';
-
-import { GovernanceContracts, GovernanceContractAddresses } from './contracts';
-import { Call, normalizeCall, associateCalls } from './utils';
-
-export type Governor = {
-  domain: number;
-  identifier: string;
-};
-
-export class AbacusGovernance extends AbacusApp<
-  GovernanceContractAddresses,
-  GovernanceContracts
-> {
-  readonly calls: Map<number, Readonly<Call>[]>;
-
+import { MultiProvider } from '../provider';
+import { ChainName, ChainSubsetMap } from '../types';
+import { objMap, promiseObjAll } from '../utils';
+import { GovernanceAddresses } from './contracts';
+import { environments } from './environments';
+
+type Environments = typeof environments;
+type EnvironmentName = keyof Environments;
+
+export class AbacusGovernance<
+  Networks extends ChainName = ChainName,
+> extends AbacusApp<GovernanceContracts, Networks> {
   constructor(
-    addresses: Partial<Record<ChainName, GovernanceContractAddresses>>,
+    networkAddresses: ChainSubsetMap<Networks, GovernanceAddresses>,
+    multiProvider: MultiProvider<Networks>,
   ) {
-    super();
-    const chains = Object.keys(addresses) as ChainName[];
-    chains.map((chain) => {
-      this.registerDomain(domains[chain]);
-      const domain = this.resolveDomain(chain);
-      this.contracts.set(domain, new GovernanceContracts(addresses[chain]!));
-    });
-    this.calls = new Map();
-  }
-
-  /**
-   * Returns the governors of this abacus deployment.
-   *
-   * @returns The governors of the deployment
-   */
-  async governors(): Promise<Governor[]> {
-    const governorDomains = Array.from(this.contracts.keys());
-    const governorAddresses = await Promise.all(
-      governorDomains.map((domain) =>
-        this.mustGetContracts(domain).router.governor(),
+    super(
```

Comment:
Thoughts on moving some of this to AbacusApp? Not sure if the following is possible:
```
export class AbacusApp<
  Contracts extends IAbacusContracts<any>,
  Networks extends ChainName = ChainName,
  Addresses extends AbacusAddresses,
> extends MultiGeneric<Contracts, Networks> {

constructor(networkAddresses: ChainSubsetMap<Networks, Addresses>, multiProvider: MultiProvider<Networks>, contracts: typeof Contracts) {
  super(networkAddresses, (local, addresses) => new Contracts(addresses, multiProvider.getDomainConnection(local).getConnection())
}
```

This would allow the governance constructor to just be:
```
constructor(networkAddresses, multiProvider) { 
  super(networkAddresses, multiProvider, GovernanceContracts
}

Reply:
Per our discussion you already figured this out!